source/AI/ShipAICache.h

```diff
@@ -0,0 +1,59 @@
+/* ShipAICache.h
+Copyright (c) 2022 by Hurleveur
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+*/
+
+#ifndef SHIP_AI_CACHE_H_
+#define SHIP_AI_CACHE_H_
+
+class Ship;
+
+
+
+// Class representing the AI data of a single ship. With a seperate instance
+// for every ship.
+class ShipAICache {
+public:
+	ShipAICache() = default;
+	// Construct and Load() at the same time.
+	ShipAICache(const Ship &ship);
+
+	void UpdateWeaponCache();
+
+	// Accessors and setters for AI data
+	bool IsArtilleryAI() const;
+	double ShortestRange() const;
+	double ShortestArtillery() const;
+	double MinSafeDistance() const;
+	double TurningRadius() const;
+
+
+private:
+	const Ship* ship = nullptr;
+
+	bool artilleryAI = false;
```

Comment:
```suggestion
	bool useArtilleryAI = false;
```
Perhaps also rename the method for getting this as well.